src/NuGet.Core/NuGet.Build.Tasks.Pack/NuGet.Build.Tasks.Pack.targets

```diff
@@ -36,7 +36,7 @@ Copyright (c) .NET Foundation. All rights reserved.
     <BuildOutputTargetFolder Condition="'$(BuildOutputTargetFolder)' == '' AND '$(IsTool)' == 'true'">tools</BuildOutputTargetFolder>
     <BuildOutputTargetFolder Condition="'$(BuildOutputTargetFolder)' == ''">lib</BuildOutputTargetFolder>
     <ContentTargetFolders Condition="'$(ContentTargetFolders)' == ''">content;contentFiles</ContentTargetFolders>
-    <PackDependsOn>$(BeforePack); _IntermediatePack; GenerateNuspec; $(PackDependsOn)</PackDependsOn>
+    <PackDependsOn>$(BeforePack); _IntermediatePack; GenerateNuspecNoNuspecFile; GenerateNuspecWithNuspecFile; $(PackDependsOn)</PackDependsOn>
```

Comment:
change this to be conditional on the presence of NuspecFile property and it will depend on either _GenerateNuspecUsingProjectFile or _GenerateNuspecUsingNuspecFile